modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -843,12 +846,34 @@ private Map<String, String> _addAction(
 				String scopeKey)
 		throws Exception {
 
+		Map<String, Object> properties = objectEntry.getProperties();
+
 		if (objectRelationships.isEmpty()) {
+			for (ObjectRelationship objectRelationship :
+					_objectRelationshipLocalService.getAllObjectRelationships(
+						objectDefinition.getObjectDefinitionId())) {
+
+				ObjectDefinition relatedObjectDefinition =
+					_getRelatedObjectDefinition(
+						objectDefinition, objectRelationship);
+
+				String objectRelationshipObjectFieldName =
+					_getObjectRelationshipObjectFieldName(
+						properties, objectRelationship.getName());
+
+				if (objectRelationshipObjectFieldName != null) {
+					long relatedObjectEntryId = MapUtil.getLong(
+						properties, objectRelationshipObjectFieldName);
+
+					_objectEntryService.checkModelResourcePermission(
+						relatedObjectDefinition.getObjectDefinitionId(),
+						relatedObjectEntryId, ActionKeys.UPDATE);
+				}
+			}
+
```

Comment:
@vngoliveira I'm a bit confused about this fix.

1. Why the issue does not happen when we send the `ERC`, like you showed me, but this is the only place where we check permissions for relationship?
2. Also, we shouldn't check permission in the REST layer like this. The service layer should be responsible for this.
3. Why do we need to get all relationships? This will **negatively impact performance**.
4. You don't need to use a regex. You can get the relationship object field name by doing: 
```java
ObjectField objectField = objectFieldLocalService.getObjectField(
	objectRelationship.getObjectFieldId2());

objectField.getName();
```

Please investigate how the code handles `ERC` and then follow the same procedure for `ID`.